fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/entity/FabricBlockEntityTypeBuilder.java

```diff
@@ -32,7 +32,10 @@
  * Fabric's version of BlockEntityType.Builder.
  * Removes the need to use an access widener on the BlockEntityFactory
  *
+ * @deprecated Replaced by access widener for {@link BlockEntityType.BlockEntityFactory}
+ * in Fabric Transitive Access Wideners (v1).
  */
+@Deprecated
```

Comment:
This class provides its own functionality for adding individual blocks that the `BlockEntityFactory` class does not provide.

Reply:
What's the use case, though? Can't you just create the BlockEntityType once you finished building your collection of Blocks?  
I feel like that would be better for modders to do rather than use #addBlock(s) methods.

Reply:
> This class provides its own functionality for adding individual blocks that the BlockEntityFactory class does not provide.

Imo that functionality is not worth having in fabric API. If you need it, you can easily emulate it in your mod.

Reply:
This boilerplate feature comes at barely any cost.

Reply:
Mojang's design is to ensure that blocks are initialized before its entities, and this feature may encourage bad coding behavior of messing up initialization order.

Reply:
But this doesn't allow initialising the BET before its blocks are defined anyway since the builder will only be turned into a BET once all the `addBlock`s and such have been done. (And the feature has already been accepted in an earlier PR)

Reply:
> Mojang's design is to ensure that blocks are initialized before its entities, and this feature may encourage bad coding behavior of messing up initialization order.

Uh? No?
This is done **before** the block entity type is actually built and registered.

And even then, there's some use cases where adding supported blocks after the block entity type is added is needed, especially in the case where blocks are added through analysis for the registry, which means you never know when you can finally let go and register the block entity type.

Reply:
The objective is to eventually create a new module with only the strictly necessary object builders, or migrate its contents to different modules with a clearer purpose, or a combination of both. As such, it's unnecessary to keep these APIs in the non-deprecated modules indefinitely. However, they will still be accessible in object builders v1 until mojang breaks this code.

Reply:
Regarding Mojang's design for how blocks and block entities are initialized, registry freezing allows data requiring both blocks and block entities to be calculated after both have been registered.

Reply:
We should move this addition functionality to something like `FabricBlockEntityTypeBuilderAccess`

Reply:
It's not really a matter of whatever Mojang wants, it's more about not keeping a class in fabric api just for this specific use case. It was originally introduced before TAWs were available, but now we can re-evaluate that decision.